### ISSUE TITLE
paper-input: Provide "really" isInvalid even if not touched 

### ISSUE THREAD

**Issue by craghavendra21** [miguelcobain/ember-paper#336]
Hi,

My scenario is 
I have a page in which I have more than 10 input control with required field validation.
I have submit button which will not be enabled until all the values are entered.
But when form loads all the **isInvalid** variable from input controller will be **false** so for disabling button I have to create computed property for verifying each values is entered or not, but this code you have already written.

So if you set **isInvalid** even the control is not touched (even if **isTouched** value is false).

I can understand that you are handling showing error message on **isInvalid** condition like bellow in the {{paper-input}} template

`<div id="error-{{inputElementId}}" ng-message="{{ng-message}}" class="{{if isInvalid 'ng-enter ng-enter-active' 'ng-leave ng-leave-active'}}">{{errortext}}</div>`

So instead of this if you change the code to this

`<div id="error-{{inputElementId}}" ng-message="{{ng-message}}" class="{{if isTouched 'ng-enter ng-enter-active' 'ng-leave ng-leave-active'}} {{if isInvalid 'ng-value-invalid' 'ng-value-valid'}}">{{errortext}}</div>`

And show the error message only if the class is “`ng-enter ng-enter-active ng-value-invalid`”

Then we can handle many things ouside. enabling or dissabling other properties based on this and so on.

**Comment by DanChadwick** [comment]
1. I infer from the code you quoted that you are using the master branch. This branch won't receive further development as all work is currently focused on an effort to deliver version 1.0. That is happening right now in branch `wip/v1.0.0`.

2. In `wip/v1.0.0`, validation handling has changed significantly and may or may not be more to your liking. You can try this branch, but be aware that not all component have been adapted to the new stylesheets and architecture. There is an issue tracking the status.

3. I *think* that you would like to receive indication whether a `{{paper-input}}` is "really" invalid. We give the user an opportunity to touch the field before displaying validation messages. This avoids the user seeing a blank form with 10 "This is required." messages. We could expose a "really" invalid boolean.

4. It might be easier for you to integrate with something like `ember-cp-validations` and not use `ember-paper`'s built-in validation.

Hope this helps.

**Comment by craghavendra21** [comment]
Thanks for your reply,

Yes, i wanted to know whether a {{paper-input}} is "really invalid or not" on form load itself so that i can disable the submit button if its invalid.

I agree that if validation messages shown in form load itself it will look odd. so code in the `wip/v1.0.0` branch is somthing like bellow, 

```
{{#if isInvalid}}
    <div class="md-input-messages-animation md-auto-hide" ng-messages>
      {{#each validationErrorMessages as |error index|}}
        <div
          id="error-{{inputElementId}}-{{index}}"
          class="paper-input-error ng-enter ng-enter-active" ng-message>
          {{error.message}}
        </div>
      {{/each}}
    </div>
{{/if}}
```

is that possible to make above code to look like bellow so that we will come to know whether {{paper-input}} is realy valid or not and also it will not show error messages when form load instead it will show only after user intract with the {{paper-input}}

```
{{#if isTouched}}
	{{#if isInvalid}}
		<div class="md-input-messages-animation md-auto-hide" ng-messages>
		  {{#each validationErrorMessages as |error index|}}
			<div
			  id="error-{{inputElementId}}-{{index}}"
			  class="paper-input-error ng-enter ng-enter-active" ng-message>
			  {{error.message}}
			</div>
		  {{/each}}
		</div>
	{{/if}}
{{/if}}
```

or

```
{{#if isTouched}}
	{{#unless hideAllMessages}}
	  <div class="md-errors-spacer">
		{{#if maxlength}}
		  <div class="md-char-counter">{{renderCharCount}}</div>
		{{/if}}
	  </div>
	  {{#if isInvalid}}
		<div class="md-input-messages-animation md-auto-hide" ng-messages>
		  {{#each validationErrorMessages as |error index|}}
			<div
			  id="error-{{inputElementId}}-{{index}}"
			  class="paper-input-error ng-enter ng-enter-active" ng-message>
			  {{error.message}}
			</div>
		  {{/each}}
		</div>
	  {{/if}}
	{{/unless}}
{{/if}}
```

**Comment by DanChadwick** [comment]
You won't have the right styling for the input element itself with your proposed changes. Try using:
```javascript
  isReallyInvalid: computed('validationErrorMessages.length', 'isNativeInvalid', function() {
    return (this.get('validationErrorMessages.length') || this.get('isNativeInvalid'));
  }),
```
You can use `reopen` to add this to `paper-input`, or define your own subclass of `paper-input`.

**Comment by DanChadwick** [comment]
Closing in favor of #226, which appears to be seeking a solution to the same problem. Please re-open if I'm in error.
